packages/mdx/mdx-ast-to-mdx-hast.js

```diff
@@ -29,12 +29,12 @@ module.exports = function mdxAstToMdxHast() {
         if (lang) {
           props.className = ['language-' + lang]
         }
-
-        props.metaString = node.lang && node.lang.replace(langRegex, '').trim()
+        
+        const metaString = node.lang && node.lang.replace(langRegex, '').trim()
```

Comment:
Could you instead tell it to pass `metaString` as a prop only if it's not an empty string? By exposing the raw string we will give more power to the user in the long run. If people use the infostring, extracted props will also be unrecognized so the user would have to define the `code` key of `components` anyway and deal with those anyway.

```suggestion
        const metaString = node.lang && node.lang.replace(langRegex, '').trim()
        props.metaString = metaString !== '' ? metaString : null
```

(I'm not sure if this is how suggestions work with multiple lines.)

Reply:
This PR is trying to _avoid_ setting the `metaString` prop as it's illegal JSX. If you want the `metastring` prop to be rendered in the DOM it should be `props.metastring` (notice the lowercase `s`), should I change it to that?

Reply:
The goal is to get rid of the `metaString` warning in this case:

~~~md
```js
const a = 5;
```
~~~

because in the cases where the string actually contains something:

~~~md
```js exec=true
```
~~~

the prop `exec` will also get passed to tag `<code>`. React won't complain, but I can't imagine many cases where you would want this behavior, you probably want to intercept `exec` as well as `metaString` and not pass it forward:

```js
const components = {
  code: ({ metaString, exec, ...props }) => {
    // do something based on exec or metaString
    return <code {...props} /> // no React warning
  },
}
```

Infostring is a rarely used features, most people getting this warning are not using it in the first place. When they are, I don't think it's much hassle to prevent passing it to the DOM node, I think the warning even helps with that by telling you that you probably want to create a `code` component.

The warning doesn't help when `metaString` is empty, which is most often the case, so to me it makes sense to omit it only in that case.

What do you think?

Reply:
Actually, I think I see what you mean. Give me a day to think about it.